scene/3d/physics/collision_shape_3d.h

```diff
@@ -32,12 +32,14 @@
 
 #include "scene/3d/node_3d.h"
 #include "scene/resources/3d/shape_3d.h"
+#include "scene/resources/physics_material.h"
 
 class CollisionObject3D;
 class CollisionShape3D : public Node3D {
 	GDCLASS(CollisionShape3D, Node3D);
 
 	Ref<Shape3D> shape;
+	Ref<PhysicsMaterial> physics_material;
```

Comment:
Looking at the relationship between `CollisionShape*D` and `CollisionObject*D` more closely, it would seem to me that this is not quite correct, and should instead be implemented much in the same way that the `disabled` property is currently implemented, where it should live in `CollisionObject3D::ShapeData` instead, and there should be `shape_owner_*` methods (that are exposed to scripting) like `shape_owner_set_physics_material` and such.

I frankly don't quite understand the purpose of all this "shape owner" stuff, and why there isn't just one shape owner per object, but again, this new physics material sits alongside the `disabled` property in `CollisionShape*D`, so there's no reason why it shouldn't be implemented in the same way.

Reply:
We have several properties that do nothing on certain subclasses (Light3D is an example of this with some of its properties only being usable with OmniLight3D and SpotLight3D), so I guess it's fine to do this way.